src/gfx/geometries/ExtrudedObjectsGeometry.js

```diff
@@ -85,13 +85,66 @@ class ExtrudedObjectsGeometry extends ChunkedObjectsGeometry {
         positions[vtxIdx + 1] = point.y;
         positions[vtxIdx + 2] = point.z;
 
-        tmpPrev.subVectors(point, prevPt).normalize();
-        tmpNext.subVectors(point, nextPt).normalize();
-        tmpPrev.add(tmpNext).normalize();
+        // Counting normals:
+        // - No slope
+        //   Radius doesn't change throught part => normals are parallel with the plane contains section points
+        //   normal = vToPrevPointInSection + vToNextPointInSection (all vectors are scaled for being 1 in length)
+        // - Slope
+        //   Radius changes throught part => normals aren't parallel with the plane contains section points
+        //   normal = vTangentInSectionPlane x vToSuchPointInPrevSection (all vectors are scaled for being 1 in length)
+        if (hasSlope) {
+          if (i !== 1) {
+            // zero and first sections are equal so we will have tmpNext = 0 for the first section
+            // so we count normals with another way for it
+            tmpPrev.subVectors(prevPt, nextPt).normalize();
+            tmpNext.x = point.x - positions[vtxIdx - ptsCount * 3 + 0];
```

Comment:
You use `vtxIdx - ptsCount * 3` several times, but I advise you to calculate it once and then only increment it.